### ISSUE TITLE
Core Infrastructure Initiative (CII) Best Practices

### ISSUE THREAD

**Issue by ypid** [QubesOS/qubes-issues#2167]
#### Proposal

Someone closely involved with the project could go thought the criteria and keep them up-to-date. I think you have a pretty good chance of meeting most [criteria](https://bestpractices.coreinfrastructure.org/criteria) already :wink:

#### References

* https://bestpractices.coreinfrastructure.org
* https://twit.tv/shows/floss-weekly/episodes/389

**Comment by rugk** [comment]
Good idea, so in case you need a way to tick the criteria:

* [ ] homepage_url
* [ ] description_good
* [ ] interact
* [ ] contribution
* [ ] contribution_requirements
* [ ] floss_license
* [ ] floss_license_osi
* [ ] license_location
* [ ] documentation_basics
* [ ] sites_https
* [ ] english
* [ ] repo_public
* [ ] repo_track
* [ ] repo_distributed
* [ ] repo_track
* [ ] version_unique
* [ ] version_semver
* [ ] version_tags
* [ ] release_notes
* [ ] release_notes_vulns
* [ ] report_process
* [ ] report_tracker
* [ ] report_responses
* [ ] enhancement_responses
* [ ] report_archive
* [ ] vulnerability_report_process
* [ ] vulnerability_report_private
* [ ] build
* [ ] build_common_tools
* [ ] build_floss_tools
* [ ] test
* [ ] test_invocation
* [ ] test_most
* [ ] test_continuous_integration
* [ ] test_policy
* [ ] tests_are_added
* [ ] tests_documented_added
* [ ] warnings
* [ ] warnings_fixed
* [ ] warnings_strict
* [ ] know_secure_design
* [ ] know_common_errors
* [ ] crypto_published
* [ ] crypto_call
* [ ] crypto_floss
* [ ] crypto_keylength
* [ ] crypto_working
* [ ] crypto_weaknesses
* [ ] crypto_pfs
* [ ] crypto_password_storage
* [ ] crypto_random
* [ ] delivery_mitm
* [ ] delivery_unsigned
* [ ] vulnerabilities_fixed_60_days
* [ ] vulnerabilities_critical_fixed
* [ ] no_leaked_credentials
* [ ] static_analysis
* [ ] static_analysis_common_vulnerabilities
* [ ] static_analysis_fixed
* [ ] static_analysis_often
* [ ] dynamic_analysis
* [ ] dynamic_analysis_enable_assertions
* [ ] dynamic_analysis_fixed
* [ ] installation_common
* [ ] build_reproducible
* [ ] crypto_used_network
* [ ] crypto_tls12
* [ ] crypto_certificate_verification
* [ ] crypto_verification_private
* [ ] hardened_site
* [ ] hardening

So please tick the things you already meet.

**Comment by andrewdavidwong** [comment]
Thanks, @rugk. I removed some duplicates, added some missing criteria, and filled in all the ones I know we meet (or are N/A).

**Comment by andrewdavidwong** [comment]
Also added notes and messages for @marmarek.

**Comment by ypid** [comment]
@rugk Great idea, thanks!

You could run the following on your comment above to link each criteria to it’s corresponding description with:

```Shell
 sed --regexp-extended 's~^(-\s+\[[\sx]\]\s+)(\w+\>)~\1[\2](https://github.com/linuxfoundation/cii-best-practices-badge/blob/master/doc/criteria.md#\2)~;'
```

**Comment by rugk** [comment]
@ypid Good idea, your regexp did not work, but I've created and used [a similar one](https://regex101.com/r/1UNq5j/1).

**Comment by jpouellet** [comment]
The static and dynamic analysis criteria should be addressed by [this GSoC project](https://www.qubes-os.org/gsoc/#formally-analyze-how-untrusted-inputs-propagate-through-the-qubes-code-base) (if @paraschetal's proposal is accepted and successful)
